### PR TITLE
[WIP] Bug/381 sortupdate event index wrong

### LINKED ISSUE #281: oldindex retains original index after sortupdate with multiple moves.
Greetings,

e.detail.oldindex seems to retain the original index of the item that was moved if an item is moved more than once within a list.  Note that e.detail.oldElementIndex does appear to be updated with each move.  

I am working with items in a single list.  Observed this behavior in Firefox and Chrome, have not tried others.

Not sure if this is by design as I could see it being useful referring back to what the lists original order was but it does not sound like it from the documentation which simply states that it contains the old index of the dragged element (considering only list items).

If people do want the original order retained, perhaps an originalIndex property?

### PR DESCRIPTION
**Pull request by mfeherpataky**
Following conversation in #381, also fixes #281.

Tested with linked lists, also with copy-only lists.
Not sure if all the cases were covered, so it needs some more manual tests.

--

FYI - Automatic tests are failing for ghost handling. 
Not sure if this was after a rebase on the new master, but I suspect it was as no ghost code was altered.

Unfortunately for the next few days will not have any free time to continue.

TL;DR - needs manual tests and if works as expected - update to documentation & automated tests code

### CODE REVIEW

**Comment by lukasoppermann** [comment]
I did not have time to test it yet, but having a look at the code it looks very good. 👍 
I am currently using `sortableContainer` for what you refer to as `sortableList` in the serialize function. I think Container might be better, because you can technically sort tables and grids as well. What do you think?

I think we need to add some development notes in some files and and link those in the readme, which should include naming, etc. I will start on this but if you have ideas for what you think would make working on it easier, please let me know. For e.g. concerning the setup and typescript.

**Comment by mfeherpataky** [comment]
Yeap, I will refer to `sortableContainers` - that makes sense, especially as that's how we are naming them in the event responses.

I will come back to in on the weekend.
Did you find anything else I should fix/look into for this PR?

edit: Also - I will use `dragenter`, instead of `dragover` to fetch `itemsBeforeUpdate` for destinationContainer. The former is triggered only once and the latter every couple 100ms, with the same result. Just read specs on MDN 😄.

File: src/html5sortable.ts [hunk 1]
```diff
@@ -324,37 +339,58 @@ export default function sortable (sortableElements, options: object|string|undef
...
+          item: dragging
         }
       }))
     })
-    // Handle drag events on draggable items
+
+    /*
+     We are capturing targetSortable data with 'dragover' event
+    */
+    _on(sortableElement, 'dragover', (e) => {
```
Comment (mfeherpataky) [inline src/html5sortable.ts]: That will be changed to dragenter - much cheaper to run

**Comment by lukasoppermann** [comment]
Hey, good catch! 👍 
The only thing apart from the `dragenter`, that I can think of, is updating the `readme` and maybe improving the examples so that testing this is easier.

I did check it out and it worked quite well in my test, very neat.

**Comment by mfeherpataky** [comment]
hi @lukasoppermann 

Things done:
1. Rebased onto the latest master,
2. Renamed local variables to sortableContainer - from sortableList
3. Using `dragenter` to fire off less events when getting targetContainer itemsBeforeUpdate
4. Fixed an issue, where in the originContainer the itemsBeforeUpdate still carried placeholder
5. Updated README and docs/index.html to use new event object in console.log

--

However, I am a bit at a loss now with the tests results.
There are some local errors I get (around Ghost and Dragging) that never show up in Travis and vice-versa (export/import issues). To be honest, I cannot figure the tests out and could not since the beginning.

EDIT: Tests are passing now, though the early error messages in Travis had little - if anything - to do with the code I changed 😕.

**Comment by mfeherpataky** [comment]
I found still one issue with this branch - if `dragend` is called second (in cases of a successful drop) then it still attempts to remove the placeholder. Since it is not checking if the placeholder is still there, console errors out on calling .remove() on undefined.

Will fix it and push again

**Comment by lukasoppermann** [comment]
Very nice, thank you for the great work on this. 😃 This improves the events quite a lot!
